### PR TITLE
add missing id field to context

### PR DESCRIPTION
**Pull request by guykr**

### CODE REVIEW

**Review by ori-moisis — APPROVED** [review]
approving so we can unblock master.
not sure this creates correct scenarios, seems like the ID is always of the top level element, are we sure that is the correct value for all these cases? @alonstern you would probably be the most familiar with this

**Comment by alonstern** [comment]
> approving so we can unblock master.
> not sure this creates correct scenarios, seems like the ID is always of the top level element, are we sure that is the correct value for all these cases? @alonstern you would probably be the most familiar with this

I think where the `ctx.ref.element` in the tests is a field, such as line 105,  `ctx.ref.id` should also be the elemID of the field (though it won't make much difference to the tests since  `getServiceUrl` does not use `ctx.ref.id`)

**Comment by ori-moisis** [comment]
> > approving so we can unblock master.
> > not sure this creates correct scenarios, seems like the ID is always of the top level element, are we sure that is the correct value for all these cases? @alonstern you would probably be the most familiar with this
> 
> I think where the `ctx.ref.element` in the tests is a field, such as line 105, `ctx.ref.id` should also be the elemID of the field (though it won't make much difference to the tests since `getServiceUrl` does not use `ctx.ref.id`)

we should still fix it even if right now it doesn't make a difference.
we don't want tests to create unrealistic scenarios because they make it harder to maintain in the long run (when the test gets broken it is much harder to understand if the failure it real or not, this tends to be the biggest time waster in significant refactors from my experience with our codebase)

**Comment by guykr** [comment]
@alonstern  can you give an example of how this change might look like?

File: packages/lang-server/test/service_url.test.ts [hunk 1]
```diff
@@ -129,6 +138,7 @@ describe('getServiceUrl', () => {
       element: type.fields.fieldName,
       path: [],
       isList: false,
+      id: elemID,
```
Comment (alonstern) [inline packages/lang-server/test/service_url.test.ts:141]:
```suggestion
      id: type.fields.fieldName.elemID,
```

**Comment by alonstern** [comment]
> @alonstern can you give an example of how this change might look like?

Sure, added in another comment

File: packages/lang-server/test/service_url.test.ts (hunk 1 above)
Comment (guykr) [reply packages/lang-server/test/service_url.test.ts:141]: thanks! is that the only place?

Comment (alonstern) [reply packages/lang-server/test/service_url.test.ts:141]: I think also line 108

**Pull request merged**